stream/src/public_key/connection.rs

```diff
@@ -101,25 +122,22 @@ impl<Si: Sink, St: Stream> Connection<Si, St> {
 
         // Generate shared secret
         let secret = x25519::new(&mut context);
-        let ephemeral = x25519_dalek::PublicKey::from(&secret);
 
         // Send handshake
-        let timestamp = context.current().epoch_millis();
-        let msg = create_handshake(
+        let msg = handshake::Signed::sign(
             &mut config.crypto,
             &config.namespace,
-            timestamp,
-            peer.clone(),
-            ephemeral,
-        )?;
+            handshake::Info::<C>::new(peer.clone(), &secret, context.current().epoch_millis()),
```

Comment:
imo it is nice to give timestamp its own variable (given this function signature is already complex).